### PR TITLE
Client streaming cancellation test.

### PR DESCRIPTION
**Pull request by a11r**
This test exposes a crash in the C-core due to incorrect expectation that a message is always returned by the server.
Also fixed a corresponding expectation in stream.h, but the underlying issue in C-core remains (to be fixed by ctiller :)

### CODE REVIEW

**Comment by ctiller** [comment]
Assuming the DO NOT MERGE can be lifted now?

**Comment by a11r** [comment]
Removed the do not merge. Triggering travis reruns.

File: test/cpp/end2end/end2end_test.cc [hunk 1]
```diff
@@ -491,6 +491,27 @@ TEST_F(End2endTest, ServerCancelsRpc) {
   EXPECT_TRUE(s.details().empty());
 }
 
+// Client cancels request stream after sending two messages
+  TEST_F(End2endTest, ClientCancelsRequestStream) {
```
Comment (ctiller) [inline test/cpp/end2end/end2end_test.cc]: Indentation

**Comment by ctiller** [comment]
Can you merge this with head?

**Comment by a11r** [comment]
All Something broke in my rebase. I will retry this PR with a fresh branch.

**Pull request closed without merging**